aggregator_api/src/lib.rs

```diff
@@ -0,0 +1,814 @@
+        pub(crate) collector_auth_tokens: Vec<String>,
+        pub(crate) aggregator_hpke_configs: HashMap<HpkeConfigId, HpkeConfig>,
+    }
+
+    impl From<&Task> for TaskResp {
+        fn from(task: &Task) -> Self {
+            let encoded_verify_keys: Vec<_> = task
+                .vdaf_verify_keys()
+                .iter()
+                .map(|key| URL_SAFE_NO_PAD.encode(key))
+                .collect();
+            let encoded_aggregator_auth_tokens: Vec<_> = task
+                .aggregator_auth_tokens()
+                .iter()
+                .map(|token| URL_SAFE_NO_PAD.encode(token))
+                .collect();
+            let encoded_collector_auth_tokens: Vec<_> = task
+                .collector_auth_tokens()
+                .iter()
+                .map(|token| URL_SAFE_NO_PAD.encode(token))
+                .collect();
+            let aggregator_hpke_configs: HashMap<_, _> = task
+                .hpke_keys()
+                .iter()
+                .map(|(&config_id, keypair)| (config_id, keypair.config().clone()))
+                .collect();
+
+            Self {
+                task_id: *task.id(),
+                aggregator_endpoints: task.aggregator_endpoints().to_vec(),
+                query_type: *task.query_type(),
+                vdaf: task.vdaf().clone(),
+                role: *task.role(),
+                vdaf_verify_keys: encoded_verify_keys,
+                max_batch_query_count: task.max_batch_query_count(),
+                task_expiration: *task.task_expiration(),
+                report_expiry_age: task.report_expiry_age().cloned(),
+                min_batch_size: task.min_batch_size(),
+                time_precision: *task.time_precision(),
+                tolerable_clock_skew: *task.tolerable_clock_skew(),
+                collector_hpke_config: task.collector_hpke_config().clone(),
+                aggregator_auth_tokens: encoded_aggregator_auth_tokens,
+                collector_auth_tokens: encoded_collector_auth_tokens,
+                aggregator_hpke_configs,
+            }
+        }
+    }
+
+    #[derive(Serialize)]
+    pub(crate) struct GetTaskMetricsResp {
+        pub(crate) reports: u64,
+        pub(crate) report_aggregations: u64,
+    }
+}
+
+trait ConnExt {
+    fn task_id_param(&self) -> Result<TaskId, Status>;
+}
+
+impl ConnExt for Conn {
+    fn task_id_param(&self) -> Result<TaskId, Status> {
+        TaskId::from_str(self.param("task_id").ok_or_else(|| {
+            error!("No task_id parameter");
+            Status::InternalServerError
+        })?)
+        .map_err(|err| {
+            warn!(err = %err, "Couldn't parse task_id parameter");
+            Status::BadRequest
+        })
+    }
+}
+
+fn instrumented<H: Handler>(handler: H) -> InstrumentedHandler<H> {
+    InstrumentedHandler(handler)
+}
+
+struct InstrumentedHandler<H>(H);
+
+#[async_trait]
+impl<H: Handler> Handler for InstrumentedHandler<H> {
+    async fn run(&self, conn: Conn) -> Conn {
```

Comment:
This might be a footgun in trillium that I should document, but wrapping a handler and only calling run may break contained handlers that expect to have before_send called. The api handler does some stuff in before_send — specifically setting a 200 status if the response body was set and there is not another status set. That behavior might not be necessary, but it is probably safest to delegate (and optionally instrument) `Handler::before_send` as well. You may want to delegate `Handler::name` to the inner type as well

Reply:
Hm! Given how useful wrapping handlers can be, I think figuring out & documenting the correct semantics for wrapping a handler is pretty important. (or, even further, removing the footguns to make writing a wrapping handler as easy as any other handler)

Is it necessary/useful to override the remaining methods? There are a couple of methods around upgrades, and an init function -- I went ahead and overrode those too.